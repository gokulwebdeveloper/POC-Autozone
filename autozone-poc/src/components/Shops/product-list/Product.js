import React from 'react';

const Product = (props) => {
    const detailsPath = "/single-product-details?id="+props.id;
    let imageId = props.id;
    
    if(imageId > 9){
        imageId = imageId % 9;
        if(imageId == 0){
            imageId = 9;
        }
    }

    let imgpath="../../../../src/utlis/img/product-img/product-"+imageId+".jpg";
    return(
        <div className="col-12 col-sm-6 col-lg-4"> 
            <div className="single-product-wrapper">
                
                <div className="product-img">
                    <img src={(props.product_image)} alt={""} />
                    {console.log('Image Id '+ imgpath)}
                    {/* <img src={imgpath} alt={""} /> */}
                    
                    <img className="hover-img" src={(props.product_image)} alt="" />
                    
                    <div className="product-badge offer-badge">
                        <span>{props.offer_badge}</span>
                    </div>
                    
                    <div className="product-favourite">
                        <a href="#" className="favme fa fa-heart"></a>
                    </div>
                </div>
                
                <div className="product-description">
                    <span>{props.product_brand}</span>
                    <a href={detailsPath} >
                        <h6>{props.product_name}</h6>
                    </a>
                <p className="product-price">
                    <span className="old-price">${props.original_price}</span> ${props.offer_price}
                    {/* <span className="old-price">$125</span> $95.65 */}
                    </p>

                    
                    <div className="hover-content">
                        
                        <div className="add-to-cart-btn">
                            <a href="#" className="btn essence-btn">Add to Cart</a>
                        </div>
                    </div>
                </div>
            </div>
        </div>
        
                            
    );
};

export default Product;