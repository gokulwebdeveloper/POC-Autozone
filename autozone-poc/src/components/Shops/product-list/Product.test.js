import React from 'react';
import renderer from "react-test-renderer";
import { shallow, render } from "enzyme";
import Product from './Product';

let container = null;
beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
});

describe("Render Product Component", () => {
    const product = shallow(<Product />);
    it("Renders without crashing", ()=> {
        shallow(<Product />);
    });

    it("set product image with source", () => {
        expect(product.find("img").at(0)).toBeTruthy();
    });

    it("SnapShot Match", () => {
        const page = renderer.create(<Product/>).toJSON();
        expect(page).toMatchSnapshot();
    });

    it("check offer badge", () => {
        expect(product.find("span").first().text()).toBe("");
    });

    it("check favorite product", () => {
        const productPage = render(<Product />)
        expect(product.find("a").at(0)).toBeTruthy();
        // expect(productPage.getByTestClass('favme fa fa-heart')).toHaveStyle()
        // expect(product.find("a").first().text()).toBe("");
    });

    it("check product description", () => {
        expect(product.find("span").at(1).text()).toBe("");
    });

    it("check product price", () => {
        expect(product.find("span").at(2).text()).toBe("$");
    });

    
});

afterEach(() => {
    container.remove();
    container = null;
  });