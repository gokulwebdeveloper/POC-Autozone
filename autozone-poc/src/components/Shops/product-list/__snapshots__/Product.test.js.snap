// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Product Component SnapShotÂ Match 1`] = `
<div
  className="col-12 col-sm-6 col-lg-4"
>
  <div
    className="single-product-wrapper"
  >
    <div
      className="product-img"
    >
      <img
        alt=""
      />
      <img
        alt=""
        className="hover-img"
      />
      <div
        className="product-badge offer-badge"
      >
        <span />
      </div>
      <div
        className="product-favourite"
      >
        <a
          className="favme fa fa-heart"
          href="#"
        />
      </div>
    </div>
    <div
      className="product-description"
    >
      <span />
      <a
        href="single-product-details.html"
      >
        <h6 />
      </a>
      <p
        className="product-price"
      >
        <span
          className="old-price"
        >
          $
        </span>
         $
      </p>
      <div
        className="hover-content"
      >
        <div
          className="add-to-cart-btn"
        >
          <a
            className="btn essence-btn"
            href="#"
          >
            Add to Cart
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
